proposals/type-aliases.md

```diff
@@ -2,27 +2,54 @@
 
 * **Type**: Design proposal
 * **Author**: Dmitry Petrov
-* **Contributors**: Andrey Breslav, Stanislav Erokhin
+* **Contributors**: Andrey Breslav, Stanislav Erokhin, Vladimir Reshetnikov
 * **Status**: Under consideration
 * **Prototype**: In progress
 
-## Feedback 
+## Use cases
+
+Type aliases provide alternative names for existing types, for example:
+* Function types
+```
+typealias MyHandler = (Int, String, Any) -> Unit
 
-Discussion of this proposal is held in [this issue](https://github.com/Kotlin/KEEP/issues/4).
+typealias HtmlBuilderAction = HtmlBuilder.() -> Unit
 
-## Use cases
+typealias Predicate<T> = (T) -> Boolean
+```
+* Collections (and other generics)
+```
+typealias NodeSet = Set<Network.Node>
+
+typealias FilesTable<K> = MutableMap<K, MutableList<File>>
+```
+* Nested classes
+```
+class Outer {
+  class Nested {
+    inner class Inner
+  }
+}
 
-* Avoid repetition of types in source code without a cost of introducing extra classes.
-    * Function types
-    * Collections (and other complex generics)
-    * Nested classes
-    * ...
+typealias Something = Outer.Nested.Inner
+```
+and so on.
+
+Type aliases do not introduce new types.
+Instead, they are fully expanded, and are equivalent to the corresponding underlying types.
+
+## Type aliases and tooling
+
+IDE and compiler should be fully aware of type aliases:
+* Diagnostic messages
+* Descriptor rendering in IDE (completion, structure view, etc)
+* ...
 
 ## Restrictions
 
 Java has no notion of "type aliases" and can't see them in class member signatures.
 
-For the same reason we can't enforce "newtype"
+For the same reason we can't enforce "new type"
```

Comment:
the public discussion has show that this is reason is rather weak, and provokes unnecessary discussion. We can use proper reasoning to justify lack of support for "new type":

1. Assignment-compatible type aliases will definitely be supported, because they are needed in popular use cases (naming function types and long generic types)
2. Value classes will be supported (at some point, closer to Valhalla release), because they enable many important use cases
3. There's no point in having two ways of expressing assignment-incompatible type aliases (newtype and value classes), so we don't want to support newtype.